packages/shared/src/components/ProfileMenu.tsx

```diff
@@ -1,29 +1,35 @@
-import React, { ReactElement, useContext } from 'react';
-import dynamic from 'next/dynamic';
-import { Item } from '@dailydotdev/react-contexify';
-import Link from 'next/link';
+import React, { ReactElement, useContext, useMemo } from 'react';
 import AuthContext from '../contexts/AuthContext';
 import PowerIcon from './icons/Power';
-import { KeyReferralOutlineIcon, UserIcon, TimerIcon } from './icons';
+import { KeyReferralOutlineIcon, UserIcon } from './icons';
 import DevCardIcon from './icons/DevCard';
 import SettingsIcon from './icons/Settings';
-import { IconSize } from './Icon';
 import { useLazyModal } from '../hooks/useLazyModal';
 import { LazyModal } from './modals/common/types';
 import { useSettingsContext } from '../contexts/SettingsContext';
 import { CampaignCtaPlacement } from '../graphql/settings';
 import { ReferralCampaignKey, useReferralCampaign } from '../hooks';
 import { useAnalyticsContext } from '../contexts/AnalyticsContext';
 import { AnalyticsEvent, TargetId, TargetType } from '../lib/analytics';
+import InteractivePopup, {
+  InteractivePopupPosition,
+} from './tooltips/InteractivePopup';
+import { AllowedTags, Button, ButtonProps } from './buttons/Button';
+import { LabeledImage } from './image';
+import { webappUrl } from '../lib/constants';
 
-const PortalMenu = dynamic(
-  () => import(/* webpackChunkName: "portalMenu" */ './fields/PortalMenu'),
-  {
-    ssr: false,
-  },
-);
+interface ListItem {
+  title: string;
+  buttonProps?: ButtonProps<AllowedTags>;
+}
```

Comment:
Yeah, but that one did not involved our button component, so we can't pass the props we need.

Reply:
Should we not uniform one of them maybe?
I don't see how the end result between a context menu for instance and this differs.

Reply:
Because the context menu item has only a few defined props, this one is unique for the ProfileMenu which is not using the same base container of ContextMenu from the library. Can you share how you envision it to be? The `MenuItemProps` is kind of a bit far from what we have here:
```
export interface MenuItemProps<
  TReturn = unknown,
  TArgs extends Array<unknown> = Array<unknown>,
  TAnchorProps = AnchorHTMLAttributes<HTMLAnchorElement>,
> {
  icon: ReactElement;
  label: string;
  action?: (...args: TArgs) => TReturn;
  anchorProps?: TAnchorProps;
}
```

Reply:
Hmm, let's leave it for now then, don't want to overcomplicate this PR.
Maybe something to keep in mind for future clean up if we could have 1 generic type of menu items.